src/kibana/components/doc_viewer/doc_viewer.less

```diff
@@ -21,4 +21,24 @@ doc-viewer .doc-viewer {
   .content {
     margin-top: @padding-base-vertical;
   }
-}
\ No newline at end of file
+
+  .null {
+    color: gray;
+  }
+
+  .string {
+    color: red;
+  }
+
+  .number {
+    color: blue;
+  }
+
+  .boolean {
+    color: green;
+  }
+
+  .key {
+    color: darkblue;
+  }
+}
```

Comment:
I would go with a combination of both. Create `@string-text-color`, `@number-text-color`, etc. Point those at the success-text etc variables.